jenkins/job_client.rb

```diff
@@ -1,26 +1,34 @@
 require 'rest-client'
 require 'json'
+
 module Jenkins
   class JobClient
 
-    attr_reader :async_mode, :jenkins_url, :jenkins_user, :jenkins_token, :job_name, :job_params, :job_timeout
+    attr_reader :async_mode, :jenkins_url, :jenkins_user, :jenkins_api_token, :job_token, :proxy, :job_name, :job_params, :job_timeout
 
     DEFAULT_TIMEOUT = 30
     INTERVAL_SECONDS = 10
 
     def initialize(args)
       @jenkins_url = args['INPUT_JENKINS_URL']
       @jenkins_user = args['INPUT_JENKINS_USER']
-      @jenkins_token = args['INPUT_JENKINS_TOKEN']
+      @jenkins_api_token = args['INPUT_JENKINS_API_TOKEN']
+      @proxy = normalize_proxy(args['INPUT_PROXY'])
       @job_name = args['INPUT_JOB_NAME']
-      @async_mode = args['INPUT_ASYNC'].to_s == 'true'
-      @job_params = JSON.parse(args['INPUT_JOB_PARAMS'])
+      @job_token = args['INPUT_JOB_TOKEN']
+      @job_params = JSON.parse(args['INPUT_JOB_PARAMS'] || '{}')
       @job_timeout = args['INPUT_JOB_TIMEOUT'] || DEFAULT_TIMEOUT
+      @async_mode = args['INPUT_ASYNC'].to_s == 'true'
+    end
+
+    def normalize_proxy(proxy)
+      return proxy unless proxy
+      proxy = "http://#{proxy}" unless proxy.start_with?('http://')
```

Comment:
This should do the trick:
```suggestion
      proxy = "http://#{proxy}" unless proxy.match(/^https?:/)
```